[BACKLOG-8011] Added expand/collapse shell to takeover transformation…

wseyler opened the pull request:
… tab

wingman-pentaho commented:
Build completed
===============

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

Build Commands:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -f ui/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f ui/build.xml jacoco-integration checkstyle publish-local

Cleanup Commands:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  ui/src/org/pentaho/di/ui/spoon/ExpandedContentManager.java
  ui/src/org/pentaho/di/ui/spoon/Spoon.java
  ui/test-src/org/pentaho/di/ui/spoon/ExpandedContentManagerTest.java

Unit test coverage change:
============
org.pentaho.di.ui.spoon.ExpandedContentManager:
 * Branch New Coverage : 37.50%
 * Complexity New Coverage : 26.32%
 * Instruction New Coverage : 34.78%
 * Line New Coverage : 31.67%
 * Method New Coverage : 36.36%

org.pentaho.di.ui.spoon.Spoon:
 * Complexity Change : **-.01%**
 * Instruction Change : **-.00%**
 * Line Change :  + .04%
 * Method Change : **-.03%**

Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards
File /home/buildguy/project/ui/src/org/pentaho/di/ui/spoon/ExpandedContentManager.java

1. Message: Line has trailing spaces at Line: 56
2. Message: Line has trailing spaces at Line: 57

File /home/buildguy/project/ui/src/org/pentaho/di/ui/spoon/Spoon.java

1. Message: Line has trailing spaces at Line: 9,220
2. Message: Line has trailing spaces at Line: 9,224
3. Message: Line has trailing spaces at Line: 9,228

wseyler commented:
@mdamour1976 fixed checkstyle problems

bmorrise commented on ui/src/org/pentaho/di/ui/spoon/ExpandedContentManager.java:
```diff
@@ -0,0 +1,192 @@
+   *          the browser object to maximize. We try to take up as much of the Spoon window as possible.
+   */
+  private static void maximizeExpandedContent( Browser browser ) {
+    SashForm sash = (SashForm) browser.getParent().getParent().getParent().getParent();
```
Can we add a getter to spoon to return the SashForm so we don't have to do all these getParent() calls?

wingman-pentaho commented:
Build completed
===============

:white_check_mark: This pull request has passed all validations.

Build Commands:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -f ui/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f ui/build.xml jacoco-integration checkstyle publish-local

Cleanup Commands:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  ui/src/org/pentaho/di/ui/spoon/ExpandedContentManager.java
  ui/src/org/pentaho/di/ui/spoon/Spoon.java
  ui/test-src/org/pentaho/di/ui/spoon/ExpandedContentManagerTest.java

Unit test coverage change:
============
org.pentaho.di.ui.spoon.ExpandedContentManager:
 * Branch New Coverage : 37.50%
 * Complexity New Coverage : 26.32%
 * Instruction New Coverage : 34.78%
 * Line New Coverage : 31.67%
 * Method New Coverage : 36.36%

org.pentaho.di.ui.spoon.Spoon:
 * Complexity Change : **-.01%**
 * Instruction Change : **-.00%**
 * Line Change :  + .04%
 * Method Change : **-.03%**

wseyler commented:
@mdamour1976 @bmorrise Added support for retaining sash size on a per tab instance.  Removed fragile "getParent()" code.  Fixed problem with tabDeselected events.  Added code so that full screen sashforms weight won't be saved when spoon closes (as per conversation with Mike).

wingman-pentaho commented:
Build completed
===============

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

Build Commands:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -f ui/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f ui/build.xml jacoco-integration checkstyle publish-local

Cleanup Commands:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  ui/src/org/pentaho/di/ui/spoon/ExpandedContentManager.java
  ui/src/org/pentaho/di/ui/spoon/Spoon.java
  ui/src/org/pentaho/di/ui/spoon/delegates/SpoonJobDelegate.java
  ui/src/org/pentaho/di/ui/spoon/delegates/SpoonSlaveDelegate.java
  ui/src/org/pentaho/di/ui/spoon/delegates/SpoonTabsDelegate.java
  ui/src/org/pentaho/di/ui/spoon/delegates/SpoonTransformationDelegate.java
  ui/src/org/pentaho/xul/swt/tab/TabItem.java
  ui/src/org/pentaho/xul/swt/tab/TabSet.java
  ui/test-src/org/pentaho/di/ui/spoon/ExpandedContentManagerTest.java
  ui/test-src/org/pentaho/xul/swt/tab/TabSetTest.java

Newly Broken Tests:
======================
org.pentaho.di.ui.spoon.ExpandedContentManagerTest.testHideTransformationBrowser:
-----------------------------------------------
java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.di.ui.spoon.ExpandedContentManager.hideExpandedContent(ExpandedContentManager.java:156)
	at org.pentaho.di.ui.spoon.ExpandedContentManagerTest.testHideTransformationBrowser(ExpandedContentManagerTest.java:108)

```

Unit test coverage change:
============
org.pentaho.di.ui.spoon.delegates.SpoonJobDelegate:
 * Instruction Change : **-.00%**
 * Line Change : **-.00%**

org.pentaho.di.ui.spoon.delegates.SpoonTabsDelegate:
 * Instruction Change : **-.00%**
 * Line Change : **-.00%**

org.pentaho.di.ui.spoon.delegates.SpoonTransformationDelegate:
 * Instruction Change : **-.00%**
 * Line Change : **-.00%**

org.pentaho.di.ui.spoon.ExpandedContentManager:
 * Branch New Coverage : 31.25%
 * Complexity New Coverage : 21.05%
 * Instruction New Coverage : 20.40%
 * Line New Coverage : 15.00%
 * Method New Coverage : 27.27%

org.pentaho.di.ui.spoon.Spoon:
 * Branch Change : **-.00%**
 * Complexity Change :  + .06%
 * Instruction Change :  + .00%
 * Line Change :  + .06%
 * Method Change :  + .21%

org.pentaho.xul.swt.tab.TabItem:
 * Complexity Change : **-1.17%**
 * Instruction Change :  + .15%
 * Line Change : **-.48%**
 * Method Change : **-1.89%**

org.pentaho.xul.swt.tab.TabSet:
 * Branch Change :  + 2.89%
 * Complexity Change :  + 1.40%
 * Instruction Change :  + 3.55%
 * Line Change :  + 3.75%
 * Method Change :  + 3.45%

wingman-pentaho commented:
Build completed
===============

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

Build Commands:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -f ui/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f ui/build.xml jacoco-integration checkstyle publish-local

Cleanup Commands:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  ui/src/org/pentaho/di/ui/spoon/ExpandedContentManager.java
  ui/src/org/pentaho/di/ui/spoon/Spoon.java
  ui/src/org/pentaho/di/ui/spoon/delegates/SpoonJobDelegate.java
  ui/src/org/pentaho/di/ui/spoon/delegates/SpoonSlaveDelegate.java
  ui/src/org/pentaho/di/ui/spoon/delegates/SpoonTabsDelegate.java
  ui/src/org/pentaho/di/ui/spoon/delegates/SpoonTransformationDelegate.java
  ui/src/org/pentaho/xul/swt/tab/TabItem.java
  ui/src/org/pentaho/xul/swt/tab/TabSet.java
  ui/test-src/org/pentaho/di/ui/spoon/ExpandedContentManagerTest.java
  ui/test-src/org/pentaho/xul/swt/tab/TabSetTest.java

Unit test coverage change:
============
org.pentaho.di.ui.spoon.delegates.SpoonJobDelegate:
 * Instruction Change : **-.00%**
 * Line Change : **-.00%**

org.pentaho.di.ui.spoon.delegates.SpoonTabsDelegate:
 * Instruction Change : **-.00%**
 * Line Change : **-.00%**

org.pentaho.di.ui.spoon.delegates.SpoonTransformationDelegate:
 * Instruction Change : **-.00%**
 * Line Change : **-.00%**

org.pentaho.di.ui.spoon.ExpandedContentManager:
 * Branch New Coverage : 31.25%
 * Complexity New Coverage : 21.05%
 * Instruction New Coverage : 20.40%
 * Line New Coverage : 15.00%
 * Method New Coverage : 27.27%

org.pentaho.di.ui.spoon.Spoon:
 * Branch Change : **-.00%**
 * Complexity Change : **-.01%**
 * Instruction Change : **-.01%**
 * Line Change : **-.01%**
 * Method Change : **-.04%**

org.pentaho.xul.swt.tab.TabItem:
 * Complexity Change : **-1.17%**
 * Instruction Change :  + .15%
 * Line Change : **-.48%**
 * Method Change : **-1.89%**

org.pentaho.xul.swt.tab.TabSet:
 * Branch Change :  + 2.89%
 * Complexity Change :  + 1.40%
 * Instruction Change :  + 3.55%
 * Line Change :  + 3.75%
 * Method Change :  + 3.45%

Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards
File /home/buildguy/project/ui/src/org/pentaho/di/ui/spoon/Spoon.java

1. Message: Line has trailing spaces at Line: 578

File /home/buildguy/project/ui/test-src/org/pentaho/di/ui/spoon/ExpandedContentManagerTest.java

1. Message: Unused import - org.junit.Assert.assertEquals. at Line: 25 Column: 15
2. Message: Unused import - org.junit.Assert.assertNotNull. at Line: 27 Column: 15
